alf/environments/suite_carla.py

```diff
@@ -835,6 +874,19 @@ def __init__(self,
             success_reward (float): the reward for arriving the goal location.
             success_distance_thresh (float): success is achieved if the current
                 location is with such distance of the goal
+            collision_reward (float): the reward for collision. Note that this
+                reward is only given once at the first step of contiguous collisions.
+            max_stuck_at_collision_seconds (float): the episode will end and is
+                considerred as failure if the car is stuck at the collision for
+                so many seconds,
+            stuck_at_collision_distance (float): the car is considerred as being
+                stuck at the collision if it is within such distance of the first
+                collision location.
+            sparse_reward (bool): If False, the distance reward is given at every
+                step based on how much it moves along the navigation route. If
+                True, the distance reward is only given after moving ``sparse_reward_distance``.
+            sparse_reward_distance (float): the sparse reward is given after every
+                such distance along the route has been driven.
```

Comment:
I'm actually confused about the two comments here. What does "sparse_reward_distance" exactly do?

Reply:
This means that the the car only get reward after moving ahead so much distance. After it gets such a reward, it has to move another sparse_reward_distance to get reward.

Reply:
It it makes more sense to name it "sparse_reward_interval".